packages/sdk/src/oracle/services/transaction.ts

```diff
@@ -0,0 +1,17 @@
+import { Provider, TransactionReceipt } from "../types/ethers";
+
+export class Transaction {
+  constructor(private provider: Provider) {}
```

Comment:
nit: class name seems a bit odd since it implies that this class is an abstraction for a transaction. If it were an abstraction for a transaction, I would expect it to take in the transaction hash in the constructor rather than being used for _many_ transactions.

My minor suggestion is to do one of the following:
1. Change the name.
2. Have the hash passed in in the constructor.
3. Turn this into two utility functions that take in a provider _and_ hash in each method.

Reply:
yea good call, it doesnt really do much other than provide some utility

Reply:
including hash and provider in same function call will require some refactoring id rather not do, but i can move this to a utility where it might make more sense, but it will have to keep the provider=>hash=>result style interface for now